queries.sql

```diff
@@ -1,11 +1,67 @@
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 SELECT * FROM animals WHERE name LIKE '%mon%';
-SELECT * FROM animals WHERE date_of_birth BETWEEN '2016-12-31' AND '2019-01-01';
+SELECT * FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31';
 SELECT * FROM animals WHERE neutered = 't' AND escape_attempts < 3;
 SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = 'true';
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 SELECT * FROM animals
-SELECT * FROM animals WHERE neutered = 't';
\ No newline at end of file
+SELECT * FROM animals WHERE neutered = 't';
+
+/* Rollback practise */
+ BEGIN;
+ UPDATE animals SET species = 'unspecified';
+ ROLLBACK;
+
+/* Add species data as transaction */
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon%';
+UPDATE animals SET species='pokemon' WHERE species IS NULL;
+COMMIT;
+
+/* Delete table and rollback */
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+/* Transaction deleting 3 rows and changing all -ve weights*/
+ BEGIN;
+ DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+ SAVEPOINT SP1;
+ UPDATE animals SET weight_kg = weight_kg * -1 ;
+ SELECT * FROM animals;
+ ROLLBACK TO SP1;
+ SELECT * FROM animals;
+ UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg <0 ;
+ COMMIT;
+ SELECT * FROM animals;
+
+ /*Queries */
+
+ /* How many animals are there?*/
+
+  SELECT COUNT (*) FROM animals;
+
+ /*How many animals have never tried to escape?*/
+
+ SELECT COUNT (*) FROM animals WHERE escape_attempts = 0;
+
+ /*What is the average weight of animals?*/
+
+  SELECT AVG(weight_kg) FROM animals;
+
+ /*Who escapes the most, neutered or not neutered animals?*/
+
+  SELECT neutered, SUM(escape_attempts) FROM animals GROUP BY neutered;
```

Comment:
- [ ] It could be better to use AVG or MAX instead of SUM to improve your query. 👍🏽